pkg/extensions/tbtc/tbtc_test.go

```diff
@@ -367,27 +391,1294 @@ func TestRetrievePubkey_ContextCancelled_WithWorkingMonitoring(t *testing.T) {
+		depositAddress,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	keepSignature, err := submitKeepSignature(depositAddress, tbtcChain)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = tbtcChain.ProvideRedemptionSignature(
+		depositAddress,
+		keepSignature.V,
+		keepSignature.R,
+		keepSignature.S,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// wait a bit longer than the monitoring timeout
+	// to make sure the potential transaction completes
+	time.Sleep(2 * timeout)
+
+	expectedIncreaseRedemptionFeeCalls := 1
+	actualIncreaseRedemptionFeeCalls := tbtcChain.Logger().
+		IncreaseRedemptionFeeCalls()
+	if expectedIncreaseRedemptionFeeCalls != actualIncreaseRedemptionFeeCalls {
+		t.Errorf(
+			"unexpected number of IncreaseRedemptionFee calls\n"+
+				"expected: [%v]\n"+
+				"actual:   [%v]",
+			expectedIncreaseRedemptionFeeCalls,
+			actualIncreaseRedemptionFeeCalls,
+		)
+	}
+
+	expectedDepositRedemptionFee := new(big.Int).Mul(
+		big.NewInt(2),
+		initialDepositRedemptionFee,
+	)
+
+	actualDepositRedemptionFee, err := tbtcChain.DepositRedemptionFee(
+		depositAddress,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if expectedDepositRedemptionFee.Cmp(actualDepositRedemptionFee) != 0 {
+		t.Errorf(
+			"unexpected redemption fee value\n"+
+				"expected: [%v]\n"+
+				"actual:   [%v]",
+			expectedDepositRedemptionFee.Text(10),
+			actualDepositRedemptionFee.Text(10),
+		)
+	}
+}
+
+func TestProvideRedemptionProof_StopEventOccurred_DepositRedemptionRequested(
+	t *testing.T,
+) {
+	ctx := context.Background()
+	tbtcChain := local.NewTBTCLocalChain()
+	tbtc := newTBTC(tbtcChain)
+
+	err := tbtc.monitorProvideRedemptionProof(
+		ctx,
+		constantBackoff,
+		timeout,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tbtcChain.CreateDeposit(depositAddress)
+
+	_, err = submitKeepPublicKey(depositAddress, tbtcChain)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = tbtcChain.RedeemDeposit(depositAddress)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	initialDepositRedemptionFee, err := tbtcChain.DepositRedemptionFee(
+		depositAddress,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	keepSignature, err := submitKeepSignature(depositAddress, tbtcChain)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = tbtcChain.ProvideRedemptionSignature(
+		depositAddress,
+		keepSignature.V,
+		keepSignature.R,
+		keepSignature.S,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// wait a while before triggering the stop event because the
+	// extension must have time to handle the start event
+	time.Sleep(100 * time.Millisecond)
+
+	// invoke the action which will trigger the stop event in result
+	err = tbtcChain.IncreaseRedemptionFee(
+		depositAddress,
+		toLittleEndianBytes(big.NewInt(990)),
+		toLittleEndianBytes(big.NewInt(980)),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// wait a bit longer than the monitoring timeout
+	// to make sure the potential transaction completes
+	time.Sleep(2 * timeout)
+
+	expectedIncreaseRedemptionFeeCalls := 1
```

Comment:
Could be worth adding an explanatory comment that this call comes from `IncreaseRedemptionFee` in the test and the monitoring script is not expected to do this call. At least that's my understanding of this test.

Reply:
But we are expecting exactly one call in this test.

Reply:
Right, and my understanding is that this single call comes from `tbtcChain.IncreaseRedemptionFee` called at line 1116 in this test. I am not saying we should expect more than one call to be recorded here. It just took me a while to understand where this single call comes from so I think it makes sense to leave a comment for future-us that this call does not come from the monitoring goroutine but from this test. Unless I missed something?

Reply:
Non-blocking of course, we can address it in any other PR.